src/lightcurvelynx/graph_state.py

```diff
@@ -187,15 +187,16 @@ def copy(self):
         """
         new_state = GraphState(num_samples=self.num_samples)
         new_state.num_parameters = self.num_parameters
+        new_state.sample_offset = self.sample_offset
         for node_name, node_vars in self.states.items():
             new_state.states[node_name] = {}
             for var_name, var_value in node_vars.items():
                 if self.num_samples == 1:
                     new_state.states[node_name][var_name] = var_value
                 else:
                     new_state.states[node_name][var_name] = var_value.copy()
-        for node_name, fixed_vars in self.fixed_vars.items():
-            new_state.fixed_vars[node_name] = fixed_vars.copy()
+        for node_name, var_list in self.fixed_vars.items():
+            new_state.fixed_vars[node_name] = var_list.copy()
```

Comment:
What are `var_list` elements? Should it be `copy.deepcopy` instead?

Reply:
They are actually sets of strings (variable names). In general it shouldn't be modified and a copy would be fine. But to be safe, I will change to deepcopy.